experimental/dds/tree2/src/test/shared-tree/sharedTree.spec.ts

```diff
@@ -743,6 +743,70 @@ describe("SharedTree", () => {
 			unsubscribe1();
 			unsubscribe2();
 		});
+
+		describe("can concurrently restore and edit removed tree", () => {
+			const sb = new SchemaBuilder({ scope: "shared tree undo tests" });
+			const schema = sb.intoSchema(sb.list(sb.list(sb.string)));
+
+			for (const scenario of ["restore then change", "change then restore"]) {
+				it(`with the ${scenario} sequenced`, () => {
+					const provider = new TestTreeProviderLite(2);
+					const content = {
+						schema,
+						allowedSchemaModifications: AllowedUpdateType.None,
+						initialTree: [["a"]] as any,
+					} satisfies InitializeAndSchematizeConfiguration;
+					const tree1 = provider.trees[0].schematizeInternal(content);
+					const { undoStack: undoStack1, unsubscribe: unsubscribe1 } =
+						createTestUndoRedoStacks(tree1.checkout);
+					const tree2 = provider.trees[1].schematizeInternal(content);
+					const { undoStack: undoStack2, unsubscribe: unsubscribe2 } =
+						createTestUndoRedoStacks(tree2.checkout);
+
+					provider.processMessages();
+
+					// Validate insertion
+					validateTreeContent(tree2.checkout, content);
+
+					// edit subtree
+					tree2.root[0].insertAtEnd(["c"]);
+					provider.processMessages();
+					const contentAfterInsert = {
+						schema,
+						allowedSchemaModifications: AllowedUpdateType.None,
+						initialTree: [["a", "c"]] as any,
+					} satisfies InitializeAndSchematizeConfiguration;
```

Comment:
Oh my! I didn't realize this `validateTreeContent` function took all that stuff, that was not good advice.
Looks like other tests use `.asArray` on a view that they establish at the start of the test. Can you do something like that instead?